src/lib.rs

```diff
@@ -13,44 +13,78 @@
 
 pub use arbitrary;
 
+
+use std::os::raw::c_char;
+use std::os::raw::c_int;
+use std::ffi::CString;
+use std::{panic, ptr};
+
 extern "C" {
-    // We do not actually cross the FFI bound here.
-    #[allow(improper_ctypes)]
-    fn rust_fuzzer_test_input(input: &[u8]);
+    // This is the mangled name of the C++ function starting the fuzzer
+    fn _ZN6fuzzer12FuzzerDriverEPiPPPcPFiPKhmE(argc: *mut c_int, argv: *mut *mut *mut c_char, callback: extern fn(*const u8, usize) -> c_int );
```

Comment:
```suggestion
    #[linkage_name = "_ZN6fuzzer12FuzzerDriverEPiPPPcPFiPKhmE"]
    fn fuzzer_driver(argc: *mut c_int, argv: *mut *mut *mut c_char, callback: extern fn(*const u8, usize) -> c_int );
```

and we miiiight also need to add an extra underscore for macos to the mangled name here with a `cfg_attr` or something.